modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -1741,6 +1799,50 @@ public long maxWalSegmentSize() {
         return maxWalSegmentSize;
     }
 
+    private int getSegmentStatus(long segmentId) {
+        int idx = (int) segmentId % dsCfg.getWalSegments();
+
+        return segmentStatus.get(idx);
+    }
+
+    private boolean setSegmentStatus(long segmentId, int expected, int updated) {
+        int idx = (int) segmentId % dsCfg.getWalSegments();
+
+        boolean statusChanged = segmentStatus.compareAndSet(idx, expected, updated);
+
+        if (statusChanged)
+            log.info("Status for segment " + segmentId + " has been changed from " + segmentStatusText(expected)
+                    + " to " + segmentStatusText(updated));
+
+        return statusChanged;
+    }
+
+    private String segmentStatusText(int status) {
```

Comment:
Why not just use an `enum`?

Reply:
Initially it was enum, but code was too clunky with that.
I wanted to leave the same structure : AtomicIntegerArray. In this case, I have to convert int from enum and vice versa.